packages/synapse-react-client/src/components/TanStackTable/StyledVirtualTanStackTable.tsx

```diff
@@ -0,0 +1,123 @@
+import { VirtualItem, Virtualizer } from '@tanstack/react-virtual'
+import { noop } from 'lodash-es'
+import React, { useContext } from 'react'
+import { mergeSlotProps } from '../../utils/slots/SlotUtils'
+import StyledTanStackTable, {
+  StyledTanStackTableProps,
+} from './StyledTanStackTable'
+
+import { TrProps } from './types'
+
+type StyledVirtualTanStackTableProps<T = unknown> = Omit<
+  StyledTanStackTableProps<T, VirtualItem>,
+  'slots' | 'rows' | 'rowTransform'
+> & {
+  rowVirtualizer: Virtualizer<any, any>
+  onTableContainerScroll?: (target: EventTarget) => void
+}
+
+// Context to pass the row virtualizer to the Tr component
+const RowVirtualizerContext = React.createContext<
+  Virtualizer<any, any> | undefined
+>(undefined)
+
+/**
+ * A table row component modified to be used with @tanstack/react-virtual.
+ */
+export function VirtualizedTr<TData = unknown>(
+  props: TrProps<TData, VirtualItem>,
+) {
+  const { row: virtualItem, tableRow: _tableRow, ...rest } = props
+  const rowVirtualizer = useContext(RowVirtualizerContext)
+  return (
+    <tr
+      data-index={virtualItem.index} // needed for dynamic row height measurement
+      ref={node => rowVirtualizer?.measureElement(node)} //measure dynamic row height
+      {...rest}
+      style={{
+        ...rest.style,
+        display: 'flex',
+        position: 'absolute',
+        transform: `translateY(${virtualItem.start}px)`, //this should always be a `style` as it changes on scroll
+        width: '100%',
+      }}
+    />
+  )
+}
+
+/**
+ * Stylized table component that is compatible with @tanstack/react-table and @tanstack/react-virtual to display a
+ * virtualized table, i.e. a table that uses a virtualizer to only render the DOM nodes that contain data that is visible
+ * within the viewport. For cases where the table rows do not need to be virtualized, use {@link StyledTanStackTable}.
+ */
+export default function StyledVirtualTanStackTable<T = unknown>(
```

Comment:
New component that adds the styles / component overrides needed for virtualization to work